lib/src/common/disposable.dart

```diff
@@ -112,6 +112,12 @@ class _ObservableTimer implements Timer {
 
   @override
   bool get isActive => _timer.isActive;
+
+  @override
+  int get tick => _timer.tick; // Dart2 only
+  //int get tick => 0; // Works for Dart1 & 2, but will this break something about timers?
+  // can we re-implement the tick value correctly? Should we inject our own zone handler for
+  // timer creation to call the complete method?
```

Comment:
@georgelesica-wf Thoughts? Under dart 1 a tick of always zero would be ok since it is not used. Under Dart 2, I'm not sure what always returning zero would break in Dart 2, if anything.

Reply:
From the DAB Core library changes doc:

> The original Timer implementation ensured that all ticks of a periodic timer would execute. If the timer fell behind, it would try to catch up, but would never lose ticks. This behavior leads to big problems when a computer goes to sleep (or a Dart program isn't executed for a long time). When the Dart program is resumed, it will try to run *all* missed ticks. In Dart 2.x, it will skip the missed ticks. In return, the Timer class has a tick member that informs the program how many ticks should have passed:

```dart
/**
   * The number of durations preceding the most recent timer event.
   *
   * The value starts at zero and is incremented each time a timer event
   * occurs, so each callback will see a larger value than the previous one.
   *
   * If a periodic timer with a non-zero duration is delayed too much,
   * so more than one tick should have happened,
   * all but the last tick in the past are considered "missed",
   * and no callback is invoked for them.
   * The [tick] count reflects the number of durations that have passed and
   * not the number of callback invocations that have happened.
   */
  int get tick;
```

Reply:
we should probably either implement tick correctly, or delegate it to the underlying timer.tick. If you can think of a way to do this that is compatible with Dart 1 and Dart 2 . I'm all ears.